patsy/core/sync.py

```diff
@@ -0,0 +1,173 @@
+        # Go through the identifiers
+        for id in identifiers:
+            # Add processed file and check the path
+            self.sync_results.files_processed += 1
+            match = self.check_path(id, accessions)
+
+            if match is None:
+                # Add the identifier to the list of not found files
+                self.sync_results.files_not_found.append(id)
+                continue
+
+            # Add and update
+            if not add:
+                self.sync_results.locations_added += 1
+                continue
+
+            location = self.gateway.session.query(Location) \
+                           .filter(Location.storage_location == id,
+                                   Location.storage_provider == "ApTrust") \
+                           .first()
+
+            if location is None:
+                location = Location(
+                    storage_location=id,
+                    storage_provider="ApTrust",
+                )
+                self.gateway.session.add(location)
+                match.locations.append(location)
+                self.sync_results.locations_added += 1
+            else:
+                self.sync_results.duplicate_files += 1
+
+    def check_batch(self, bag: dict) -> tuple:
+        # Check if archive in PATSy
+        batch_name = self.parse_name(bag.get('bag_name'))
+        query = self.gateway.session.query(Batch) \
+                    .filter(Batch.name == batch_name) \
+                    .first()
+
+        # If so create tuple from the query
+        if query is not None:
+            return query.id, query.name
+
+        return ()
+
+    def process(self, **params) -> SyncResult:
+        if 'created_at__lteq' in params:
+            bags = self.get_request(Sync.OBJECT_REQUEST, per_page=10000, created_at__lteq=params['created_at__lteq'])
+        elif 'created_at__gteq' in params:
+            bags = self.get_request(Sync.OBJECT_REQUEST, per_page=10000, created_at__gteq=params['created_at__gteq'])
+        else:
+            bags = self.get_request(Sync.OBJECT_REQUEST, per_page=10000)
+
+        # Get all the objects and loop over them
+        for bag in bags:
+            in_patsy = self.check_batch(bag)
+
+            # If bag was found in PATSy, go through the files in the bag
+            if in_patsy:
+                batch_id, batch_name = in_patsy
+                print(f'Querying files from {batch_name}')
+
+                accessions = self.gateway.session.query(Accession) \
+                                 .filter(Accession.batch_id == batch_id) \
+                                 .all()
+
+                per_page = bag.get('payload_file_count')
+                object_id = bag.get('id')
+                files = self.get_request(Sync.FILE_REQUEST, intellectual_object_id=object_id, per_page=per_page)
+
+                identifiers = list(map(lambda x: x.get('identifier'), files))
```

Comment:
A more "Pythonic" way to do this kind of mapping operation is to use what is known as a [list comprehension](https://docs.python.org/3/glossary.html#term-list-comprehension):

```python
identifiers = [f.get('identifier') for f in files]
```

(This also highlights another convention that I use, it that in list comprehensions I usually use a single letter variable that matches the first letter of the list I am iterating over.)